### PR TITLE
patch for work in PR 1372

### PR DESCRIPTION
**Pull request by aerorahul**
A patch for consideration in PR1372

### CODE REVIEW

**Review by RussTreadon-NOAA — APPROVED** [review]
Changes consist of (1) adding `logger` messages, (2) replacing `self.task_config` with `config`, (3) implementing `get_berror` methods

Approve.   

Once changes are merged into [RussTreadon-NOAA:feature/refactor_jedivar](https://github.com/RussTreadon-NOAA/global-workflow/tree/feature/refactor_jedivar) I'll test them in the prufsatm.

**Comment by aerorahul** [comment]
yes.
1. is basically very important now because unlike shell where `set -x` puts out every detail of work, python does not have that.  It is upto the developer to add as many logging messages.
2, 3. is to ensure we can expand to other/newer berror methods easily.

File: ush/python/pygfs/task/atm_analysis.py [hunk 1]
```diff
@@ -156,27 +153,34 @@ def finalize(self: Analysis) -> None:
...
+        logger.info(f"Compressing {len(diags)} diag files to {atmstat}.gz")
+
         # gzip the files first
+        logger.debug(f"Gzipping {len(diags)} diag files")
         for diagfile in diags:
             with open(diagfile, 'rb') as f_in, gzip.open(f"{diagfile}.gz", 'wb') as f_out:
                 f_out.writelines(f_in)
 
         # open tar file for writing
+        logger.debug(f"Creating tar file {atmstat} with {len(diags)} gzipped diag files"
```
Comment (RussTreadon-NOAA) [inline ush/python/pygfs/task/atm_analysis.py]: Closing `)` missing.   Found this while testing `atm_analysis.py` in prufsatm.   Script failed on line 166 with `invalid syntax` error.   Added `)` and `atm_analysis.py` worked in init and run jobs.

File: ush/python/pygfs/task/atm_analysis.py [hunk 2]
```diff
@@ -186,12 +190,11 @@ def finalize(self: Analysis) -> None:
...
         gprefix = f"{self.task_config.GPREFIX}"
-        gdate = to_YMDH(self.task_config.previous_cycle)
-        gsuffix = f"{gdate}" + ".txt"
+        gsuffix = f"{to_YMDH(self.task_config.previous_cycle)}" + ".txt"
         aprefix = f"{self.task_config.APREFIX}"
-        adate = to_YMDH(self.task_config.current_cycle)
-        asuffix = f"{adate}" + ".txt"
+        asuffix = f"{to_YMDH(self.task_config.current_cycle)}" + ".txt"
 
+        logger.info(f"Copying DATA/ {gprefix}*{gsuffix} to COM/ as {aprefix}*{assuffix}")
```
Comment (RussTreadon-NOAA) [inline ush/python/pygfs/task/atm_analysis.py]: Test of atm final job aborted in prufstam due to `assuffix` being undefined.   This is a typo.   Variable name is `asuffix`.   Remove extra `s` and rerun completed without error.   Expected files copied to ROTDIR.

**Review by RussTreadon-NOAA — COMMENTED** [review]
Two typos need to be corrected before merging `atm_analysis.py` into `feature/refactor_jedivar`.   Corrected copy of `atm_analysis.py` tested in prufsam atm init, run, and final jobs.   Jobs run as expected without error.

**Comment by aerorahul** [comment]
@RussTreadon-NOAA 
I fixed the typo and the bug.

**Comment by RussTreadon-NOAA** [comment]
`atm_analysis.py` updated in prufsatm.   init, run, and final rerun.  All ran to completion with expected results.

**Review by RussTreadon-NOAA — APPROVED** [review]
Tested `atm_analysis.py` in prufsatm cycle.  Script works!

Approve.

**Comment by RussTreadon-NOAA** [comment]
FYI:  The changes in `ush/python/pygfs/task/atm_analysis.py` from `aerorahul:patch/pr1372` were committed to `RussTreadon-NOAA:feature/refactor_jedivar` at [8492ccf](https://github.com/NOAA-EMC/global-workflow/pull/1372/commits/8492ccf72a856c4c9328885742540ae37cc1b916).   

The commit at [8492ccf](https://github.com/NOAA-EMC/global-workflow/pull/1372/commits/8492ccf72a856c4c9328885742540ae37cc1b916) includes one additional change not found in this PR.  [Click here](https://github.com/NOAA-EMC/global-workflow/pull/1372#discussion_r1167353833) to view the discussion for this one additional change.

This PR may be closed unless additional changes will be committed to it for inclusion in `RussTreadon-NOAA:feature/refactor_jedivar`.

**Comment by RussTreadon-NOAA** [comment]
Close because changes in this PR already brought into `feature/refactor_jedivar`.  Branch `feature/refactor_jedivar` was merged into g-w develop on 4/18/2023 (PR #1372).

**Pull request closed without merging**